src/test/java/net/snowflake/client/jdbc/BindUploaderLatestIT.java

```diff
@@ -132,34 +95,29 @@ public void testUploadedResultsSimple() throws Exception {
     rs.next();
     assertEquals(csv2, parseRow(rs));
     assertFalse(rs.next());
+    rs.close();
+    stmt.close();
   }
 
-  // Test csv correctness, and deletion after close
   @Test
-  public void testSerializeCSVSimple() throws Exception {
-    bindUploader.upload(getBindings(conn));
-
-    // CSV should exist in bind directory until the uploader is closed
-    Path p = bindUploader.getBindDir();
-    assertTrue(Files.exists(p));
-    assertTrue(Files.isDirectory(p));
-
-    File[] files = p.toFile().listFiles();
-    assertNotNull("file must exists", files);
-    assertEquals(files.length, 1);
-
-    File csvFile = files[0];
-    try (BufferedReader br =
-        new BufferedReader(
-            new InputStreamReader(new GZIPInputStream(new FileInputStream(csvFile))))) {
-      assertEquals(csv1, br.readLine());
-      assertEquals(csv2, br.readLine());
-      assertNull(br.readLine());
+  public void testUploadStreamLargeBatch() throws Exception {
+    // create large batch so total bytes transferred are about 10 times the size of input stream
+    // buffer
+    int batchSize = 1024 * 1024;
+    SnowflakePreparedStatementV1 stmt =
+        (SnowflakePreparedStatementV1) conn.prepareStatement(dummyInsert);
+    for (int i = 0; i < batchSize; i++) {
+      bind(stmt, row1);
```

Comment:
It's better to test rows with different values instead of a constant one.

Reply:
This is doable for small batches (2 or 3 rows), but the binding data is not stored in row order; it's stored alphabetically in the files and then applied back in order later on. So testing for 1024*1024 rows, it's difficult to write asserts to match the order of the data. Hence just 1 row is tested for the large batch, but there are 2 other tests that test against smaller batches of different values.

Reply:
Do you mean the order is not guaranteed while using bulk array bind?

Reply:
Right, the upload() function does not preserve order. But then order is restored after the COPY INTO command from the internal stage to the table. So for tests that run PreparedStatement#executeBatch(), order would be preserved. But for tests that run the internal upload() function used by executeBatch(), order is alphabetical.

Reply:
We have a lot of tests in PreparedStatement1IT that test the order after using executeBatch().